### DISCUSSION TITLE
Snapshot is up to date with klipper repo?

### DISCUSSION THREAD

**Discussion by Larswa** [Desuuuu/klipper#82] [Q&A]
There was some changes to how Mainsail was calculating flow, depending on new klipper commits. https://github.com/meteyou/mainsail/issues/339

I can see that snapshot is reasonably up to date. Is it safe to use?

And thanks man ... Me and my CR10S Pro V2 love what you have done!

**Comment by Desuuuu** [comment]
The snapshot branch contains the same modifications as the master branch, with more recent upstream changes applied. It does contain the new motion_report module, so you can try it if you'd like.

It's untested though, so everything might not work with the touchscreen (there could have been a breaking change at some point).

**Reply by Larswa** [reply]
Okay .. thanks. I'll give it a try.   

Most of your changes are in the src/atsamd/t5uid1 folder, right?    Not that it would help me much. Im more infrastructure than dev, and then I'm mostly C#, so moving around in this codebase is a stretch for me. 🙄

**Reply by Desuuuu** [reply]
Here's a rough list of where most of the changes are:

* Microcontroller
  * `src/generic/t5uid1`
  * `src/atsamd/t5uid1`
  * `src/avr/t5uid1`
  * `src/lpc176x/t5uid1`
  * `src/stm32/t5uid1`

* Host
  * `klippy/extras/t5uid1`

For a complete list, you should use git.
